### PR TITLE
AR: Allow drop cap after section break

### PR DESCRIPTION
**Pull request by alinaboghiu**
## What does this change?

AR should show a drop cap after a section break. 
This is a follow up on #10005

## Why?

Drop caps should appear after section breaks like they do on DCR 

Example article: https://www.theguardian.com/news/2023/dec/21/sanctuary-i-grew-up-in-the-troubles-and-have-been-seeking-a-place-of-peace-ever-since

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| ![before][] | ![after][] |

[before]: https://github.com/guardian/dotcom-rendering/assets/1229808/dec91d9f-ff9d-43d8-8ca5-1391b8a7f951
[after]: https://github.com/guardian/dotcom-rendering/assets/1229808/7a441ee2-1fe7-4f48-9ddd-619ff0a208b8

### CODE REVIEW

File: apps-rendering/src/renderer.ts [hunk 1]
```diff
@@ -200,10 +202,14 @@ const textElement =
 				}
 
 				const isFirstParagraph = node.previousSibling === null;
+				const isAfterDinkus =
+					node.previousSibling?.previousSibling?.textContent ===
```
Comment (arelra) [inline apps-rendering/src/renderer.ts:206]: Would the dinkus (love that word :p ) always be two previous siblings away?

File: apps-rendering/src/renderer.ts [hunk 2]
```diff
@@ -200,10 +202,14 @@ const textElement =
...
+					'* * *';
```
Comment (arelra) [inline apps-rendering/src/renderer.ts:207]: Would a constant help readability here?

File: apps-rendering/src/renderer.ts [hunk 3]
```diff
@@ -177,13 +177,15 @@ const plainTextElement = (node: Node, key: number): ReactNode => {
 const shouldShowDropCap = (
```
Comment (arelra) [inline apps-rendering/src/renderer.ts:177]: I guess you saw this but icymi I noticed DCR has it's own version of this function:

https://github.com/guardian/dotcom-rendering/blob/9fcc548a54cecc6a8c17da8454a90b78487a0409/dotcom-rendering/src/components/TextBlockComponent.tsx#L106-L128

Which leads to a handy `isDinkus()` and what looks like dinkus variations (ha!):

https://github.com/guardian/dotcom-rendering/blob/9fcc548a54cecc6a8c17da8454a90b78487a0409/dotcom-rendering/src/model/enhance-dividers.ts#L4-L22

**Review by arelra — APPROVED** [review]
Some non-blocking suggestions if we need the fix out asap.

Would be good to align the dinkus logic as a follow up 👍

**Comment by arelra** [comment]
Just a general comment unrelated to this PR but it seems we have quite a few bugs crop up with drop caps and dinkuses. Does this mean we have gaps in our test coverage and therefore regressions and bugs are slipping through?

File: apps-rendering/src/renderer.ts (hunk 1 above)
Comment (alinaboghiu) [reply apps-rendering/src/renderer.ts:206]: Yes, there's always a line break in between. I followed this example: https://developer.mozilla.org/en-US/docs/Web/API/Node/previousSibling#second_example

File: apps-rendering/src/renderer.ts (hunk 3 above)
Comment (alinaboghiu) [reply apps-rendering/src/renderer.ts:177]: Yes I did, but AR has always done its own thing so I don't want to change that now :(

File: apps-rendering/src/renderer.ts (hunk 2 above)
Comment (alinaboghiu) [reply apps-rendering/src/renderer.ts]: for sure

**Pull request merged**